### PR TITLE
Eagerly fail a test suite if a step is not executable

### PR DESCRIPTION
**Pull request by amoore877**
**Pull Request check list**

- [X] Commit conforms to CONTRIBUTING.md?
- [X] Proper tests/regressions included?
- [NA] Documentation updated?

**Affected functionality**
Do not continue a test suite if a test step is not executable.

**Description of change**
Noticed in a mistake for #5386 that we don't eagerly fail out a test suite if we forgot `chmod +X` on a step: https://github.com/spiffe/spire/actions/runs/10395899838/job/28789134192?pr=5386#step:9:1001

It is likely not worthwhile to continue a test if a step is not executable. Each step of a suite should really be necessary, and it should be obvious to the contributor why the test is failing rather than letting the suite continue in a mixed state.

**Which issue this PR fixes**
N/A

### CODE REVIEW

File: test/integration/test-one.sh [hunk 1]
```diff
@@ -83,8 +83,8 @@ cp -R "${TESTDIR}"/* "${RUNDIR}/"
 run-step() {
     local script="$1"
     if [ ! -x "$script" ]; then
-        log-warn "skipping \"$script\"; not executable"
-        return
+        log-warn "Failing: \"$script\"; not executable"
+        return 1
```
Comment (amartinezfayo) [inline test/integration/test-one.sh:87]: We have the `fail-now` function that can be used to fail the test.

**Review by amartinezfayo — COMMENTED** [review]
Thank you @amoore877 for bringing this up.

File: test/integration/test-one.sh (hunk 1 above)
Comment (amoore877) [reply test/integration/test-one.sh:87]: ah I had looked at that and got mixed up on behavior of `exit` within a function :)

will fix

**Comment by amoore877** [comment]
`unit-test (ubuntu-22.04`:
```
ok  	github.com/spiffe/spire/pkg/server/api	0.023s
--- FAIL: TestAttestAgent (0.34s)
    --- FAIL: TestAttestAgent/rate_limit_fails (0.01s)
        service_test.go:3436: 
            	Error Trace:	/home/runner/work/spire/spire/pkg/server/api/agent/v1/service_test.go:3436
            	            				/home/runner/work/spire/spire/pkg/server/api/agent/v1/service_test.go:3175
            	Error:      	Received unexpected error:
            	            	EOF
            	Test:       	TestAttestAgent/rate_limit_fails
        server.go:123: Gracefully stopping gRPC server
        server.go:125: Server stopped
        server.go:127: Server serve returned <nil>
        server.go:119: Waiting for handlers to drain
        server.go:121: Handlers drained
FAIL
FAIL	github.com/spiffe/spire/pkg/server/api/agent/v1	0.932s
```

perhaps additional flake to address, since obviously this PR does not touch the SPIRE source or unit test code and thus could not cause a unit test to fail

**Comment by amoore877** [comment]
maybe `unit-test (windows) also has some flake?
```
?   	github.com/spiffe/spire/pkg/agent/plugin/nodeattestor/test	[no test files]
# github.com/google/go-tpm-tools/simulator/internal
In file included from include.c:153:
D:\golang\modcache\github.com\google\go-tpm-tools@v0.4.4\simulator\ms-tpm-20-ref\TPMCmd\tpm\src/crypt/CryptCmac.c: In function 'CryptCmacEnd':
D:\golang\modcache\github.com\google\go-tpm-tools@v0.4.4\simulator\ms-tpm-20-ref\TPMCmd\tpm\src/crypt/CryptCmac.c:164:44: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  164 |         subkey.t.buffer[subkey.t.size - 1] ^= xorVal;
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
In file included from D:\golang\modcache\github.com\google\go-tpm-tools@v0.4.4\simulator\ms-tpm-20-ref\TPMCmd\tpm\include/Global.h:79,
                 from D:\golang\modcache\github.com\google\go-tpm-tools@v0.4.4\simulator\ms-tpm-20-ref\TPMCmd\tpm\include/Tpm.h:52,
                 from D:\golang\modcache\github.com\google\go-tpm-tools@v0.4.4\simulator\ms-tpm-20-ref\TPMCmd\tpm\src/support/Global.c:57,
                 from include.c:17:
D:\golang\modcache\github.com\google\go-tpm-tools@v0.4.4\simulator\ms-tpm-20-ref\TPMCmd\tpm\include/TpmTypes.h:1350:[29](https://github.com/spiffe/spire/actions/runs/10406180086/job/28818657572#step:6:30): note: at offset -1 into destination object 'buffer' of size 16
 1350 |         BYTE                buffer[MAX_SYM_BLOCK_SIZE];
      |                             ^~~~~~
?   	github.com/spiffe/spire/pkg/common/api	[no test files]
?   	github.com/spiffe/spire/pkg/common/api/rpccontext	[no test files]
?   	github.com/spiffe/spire/pkg/common/catalog/testplugin	[no test files]
?   	github.com/spiffe/spire/pkg/common/errorutil	[no test files]
?   	github.com/spiffe/spire/pkg/common/plugin/aws	[no test files]
?   	github.com/spiffe/spire/pkg/common/plugin/gcp	[no test files]
?   	github.com/spiffe/spire/pkg/common/plugin/tpmdevid	[no test files]
?   	github.com/spiffe/spire/pkg/common/profiling	[no test files]
?   	github.com/spiffe/spire/pkg/common/sddl	[no test files]
?   	github.com/spiffe/spire/pkg/common/telemetry/agent	[no test files]
?   	github.com/spiffe/spire/pkg/common/telemetry/agent/adminapi	[no test files]
?   	github.com/spiffe/spire/pkg/common/telemetry/agent/store	[no test files]
?   	github.com/spiffe/spire/pkg/common/telemetry/common	[no test files]
?   	github.com/spiffe/spire/pkg/common/telemetry/server	[no test files]
?   	github.com/spiffe/spire/pkg/common/telemetry/agent/workloadapi	[no test files]
?   	github.com/spiffe/spire/pkg/common/version	[no test files]
panic: test timed out after 1m[30](https://github.com/spiffe/spire/actions/runs/10406180086/job/28818657572#step:6:31)s
running tests:
	TestDeleteHelp (1m30s)

goroutine 1409 [running]:
testing.(*M).startAlarm.func1()
	D:/a/spire/spire/.build/windows-x86_64/go/1.22.6/go/src/testing/testing.go:2366 +0x385
created by time.goFunc
	D:/a/spire/spire/.build/windows-x86_64/go/1.22.6/go/src/time/sleep.go:177 +0x2d

goroutine 1 [chan receive]:
testing.(*T).Run(0xc000302000, {0x1629f05?, 0xc00fe713b4?}, 0x1678d48)
	D:/a/spire/spire/.build/windows-x86_64/go/1.22.6/go/src/testing/testing.go:1750 +0x3ab
testing.runTests.func1(0xc000302000)
	D:/a/spire/spire/.build/windows-x86_64/go/1.22.6/go/src/testing/testing.go:2161 +0x37
testing.tRunner(0xc000302000, 0xc000225c70)
	D:/a/spire/spire/.build/windows-x86_64/go/1.22.6/go/src/testing/testing.go:1689 +0xfb
testing.runTests(0xc000009668, {0x1be4920, 0x12, 0x12}, {0x1?, 0xfe1e93?, 0x1bf2980?})
	D:/a/spire/spire/.build/windows-x86_64/go/1.22.6/go/src/testing/testing.go:2159 +0x445
testing.(*M).Run(0xc000246e60)
	D:/a/spire/spire/.build/windows-x86_64/go/1.22.6/go/src/testing/testing.go:2027 +0x68b
main.main()
	_testmain.go:81 +0x16c

goroutine 1433 [chan receive]:
github.com/Microsoft/go-winio.(*win32PipeListener).Close(0xc0004dc050)
	D:/golang/modcache/github.com/!microsoft/go-winio@v0.6.2/pipe.go:578 +0x7a
google.golang.org/grpc.(*listenSocket).Close(0xc0004500d8)
	D:/golang/modcache/google.golang.org/grpc@v1.65.0/server.go:815 +0x27
google.golang.org/grpc.(*Server).closeListenersLocked(...)
	D:/golang/modcache/google.golang.org/grpc@v1.65.0/server.go:1954
google.golang.org/grpc.(*Server).stop(0xc000333400, 0x0)
	D:/golang/modcache/google.golang.org/grpc@v1.65.0/server.go:1892 +0x139
google.golang.org/grpc.(*Server).Stop(...)
	D:/golang/modcache/google.golang.org/grpc@v1.65.0/server.go:1876
github.com/spiffe/spire/test/spiretest.ServeGRPCServerOnListener.func2()
	D:/a/spire/spire/test/spiretest/socketapi.go:50 +0x30
testing.(*common).Cleanup.func1()
	D:/a/spire/spire/.build/windows-x86_64/go/1.22.6/go/src/testing/testing.go:1175 +0x10f
testing.(*common).runCleanup(0xc0003024e0, 0xc0002b5d40?)
	D:/a/spire/spire/.build/windows-x86_64/go/1.22.6/go/src/testing/testing.go:1353 +0xdb
testing.tRunner.func2()
	D:/a/spire/spire/.build/windows-x86_64/go/1.22.6/go/src/testing/testing.go:1683 +0x25
testing.tRunner(0xc0003024e0, 0x1678d48)
	D:/a/spire/spire/.build/windows-x86_64/go/1.22.6/go/src/testing/testing.go:1695 +0x125
created by testing.(*T).Run in goroutine 1
	D:/a/spire/spire/.build/windows-x86_64/go/1.22.6/go/src/testing/testing.go:1742 +0x390

goroutine 1434 [select]:
github.com/Microsoft/go-winio.(*win32PipeListener).listenerRoutine(0xc0004dc050)
	D:/golang/modcache/github.com/!microsoft/go-winio@v0.6.2/pipe.go:462 +0x85
created by github.com/Microsoft/go-winio.ListenPipe in goroutine 1433
	D:/golang/modcache/github.com/!microsoft/go-winio@v0.6.2/pipe.go:536 +0x20d

goroutine 37 [syscall, locked to thread]:
syscall.SyscallN(0xc0004a4118?, {0xc0002a4270?, 0xeca180?, 0xc0004a40c0?})
	D:/a/spire/spire/.build/windows-x86_64/go/1.22.6/go/src/runtime/syscall_windows.go:544 +0x107
github.com/Microsoft/go-winio.getQueuedCompletionStatus(0x180, 0xc0001d9d60?, 0xc000[31](https://github.com/spiffe/spire/actions/runs/10406180086/job/28818657572#step:6:32)82d0?, 0xc00013f9c0?, 0xffffffff)
	D:/golang/modcache/github.com/!microsoft/go-winio@v0.6.2/zsyscall_windows.go:327 +0xbb
github.com/Microsoft/go-winio.ioCompletionProcessor(0x180)
	D:/golang/modcache/github.com/!microsoft/go-winio@v0.6.2/file.go:162 +0x85
created by github.com/Microsoft/go-winio.initIO in goroutine 15
	D:/golang/modcache/github.com/!microsoft/go-winio@v0.6.2/file.go:58 +0x5d

goroutine 1435 [sync.Mutex.Lock]:
sync.runtime_SemacquireMutex(0xc0000abe38?, 0xbf?, 0x1548420?)
	D:/a/spire/spire/.build/windows-x86_64/go/1.22.6/go/src/runtime/sema.go:77 +0x25
sync.(*Mutex).lockSlow(0xc000333560)
	D:/a/spire/spire/.build/windows-x86_64/go/1.22.6/go/src/sync/mutex.go:171 +0x15d
sync.(*Mutex).Lock(...)
	D:/a/spire/spire/.build/windows-x86_64/go/1.22.6/go/src/sync/mutex.go:90
google.golang.org/grpc.(*Server).Serve(0xc000333400, {0x17[32](https://github.com/spiffe/spire/actions/runs/10406180086/job/28818657572#step:6:33)c20, 0xc0004dc050})
	D:/golang/modcache/google.golang.org/grpc@v1.65.0/server.go:910 +0x7bc
github.com/spiffe/spire/test/spiretest.ServeGRPCServerOnListener.func1()
	D:/a/spire/spire/test/spiretest/socketapi.go:47 +0x28
created by github.com/spiffe/spire/test/spiretest.ServeGRPCServerOnListener in goroutine 14[33](https://github.com/spiffe/spire/actions/runs/10406180086/job/28818657572#step:6:34)
	D:/a/spire/spire/test/spiretest/socketapi.go:46 +0xa5
FAIL	github.com/spiffe/spire/cmd/spire-server/cli/entry	90.273s
```

someone can double-check that I'm not crazy and these are actually unrelated to the change :)

**Review by azdagron — APPROVED** [review]

**Pull request merged**